keps/20190218-cluster-resources-for-crds.md

```diff
@@ -0,0 +1,106 @@
+---
+kep-number: 0005
+title: cluster-resources-for-crds
+authors:
+  - "@runyontr"
+owners:
+  - "@runyontr"
+editor: TBD
+creation-date: 2019-02-18
+last-updated: 2019-02-19
+status: provisional
+---
+
+# cluster-resources-for-crds
+
+In order to enable the creation use of certain CRDs, some cluster scoped objects may need to be created.  This should not be
+part of the creation of a particular CRD instantiation since the deletion of that instance would remove the 
+dependency from all objects.  Allowing a Framework or FrameworkVersion to define a set of Cluster objects that are present
+to support the creation and management of CRDs would circumvent the CRD from having to create and manage the object.
+
+
+## Table of Contents
+
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+* [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+In order to enable the creation use of certain CRDs, some cluster scoped objects, or common objects, may need to be created.  This should not be
+part of the creation of a particular CRD instantiation since the deletion of that instance would remove the 
+dependency from all objects.  Allowing a Framework or FrameworkVersion to define a set of Cluster objects that are present
+to support the creation and management of CRDs would circumvent the CRD from having to create and manage the object.
+
+## Motivation
+
+### Goals
+
+* Allow the creation of common Kubernetes components in support of running a CRD.
+
+### Non-Goals
+
+* A mechanism for CRDs to create cluster scoped objects.
+
+## Proposal
+
+### User Stories [optional]
+
+
+#### Story 1
+
+A database Framework may want a Restic server deployed as part of the Framework to provide a central location for storing backups. 
```

Comment:
Would the `zookeeper` framework for `kafka` then be such an example as well or something different?

Reply:
Those COULD be if we implement backup/restore plans.